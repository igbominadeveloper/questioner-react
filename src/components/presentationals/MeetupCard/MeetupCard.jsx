import React from 'react';
import moment from 'moment';

import Image from '../Image/Image';
import { DEFAULT_MEETUP_IMAGE } from '../../../config/config';
const MeetupCard = ({ values }) => {
  return (
    <div className="meetup shadow pb-2">
      <div className="meetup-image">
        <Image url={values.images[0] || DEFAULT_MEETUP_IMAGE} />
      </div>
      <p className="meetup-title">{values.topic}</p>
      <p className="meetup-date text-light">
        <i className="fa fa-calendar" />
        {moment(values.date).format('LL')}
      </p>
      <p className="meetup-date text-light">
        <i className="fa fa-clock-o" />
        {moment(values.date).format('hh:mm a')}
      </p>
      <p className="meetup-location text-light">
        <i className="fa fa-map-marker" />
        {values.location}
      </p>
    </div>
  );
};

export default MeetupCard;
