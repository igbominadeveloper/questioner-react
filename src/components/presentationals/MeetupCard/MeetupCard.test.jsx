import React from 'react';
import { shallow } from 'enzyme';

import MeetupCard from './MeetupCard';

const props = {
  values: {
    images: [],
  },
};

describe('MeetupCard', () => {
  describe('Render', () => {
    it('renders without error', () => {
      const component = shallow(<MeetupCard {...props} />);
      expect(component).toMatchSnapshot();
    });
  });
});
