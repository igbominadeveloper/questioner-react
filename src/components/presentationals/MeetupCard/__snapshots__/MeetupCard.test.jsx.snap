// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeetupCard Render renders without error 1`] = `
<div
  className="meetup shadow pb-2"
>
  <div
    className="meetup-image"
  >
    <Image
      url="http://res.cloudinary.com/igbominadeveloper/image/upload/v1559170146/ian-schneider-66374-unsplash_afbydt.jpg"
    />
  </div>
  <p
    className="meetup-title"
  />
  <p
    className="meetup-date text-light"
  >
    <i
      className="fa fa-calendar"
    />
    June 3, 2019
  </p>
  <p
    className="meetup-date text-light"
  >
    <i
      className="fa fa-clock-o"
    />
    12:10 pm
  </p>
  <p
    className="meetup-location text-light"
  >
    <i
      className="fa fa-map-marker"
    />
  </p>
</div>
`;
